griptape/drivers/vector/astra_db_vector_store_driver.py

```diff
@@ -0,0 +1,224 @@
+
+    Attributes:
+        embedding_driver: a `griptape.drivers.BaseEmbeddingDriver` for embedding computations within the store
+        api_endpoint: the "API Endpoint" for the Astra DB instance.
+        token: a Database Token ("AstraCS:...") secret to access Astra DB. An instance of `astrapy.authentication.TokenProvider` is also accepted.
+        collection_name: the name of the collection on Astra DB.
+        environment: the environment ("prod", "hcd", ...) hosting the target Data API.
+            It can be omitted for production Astra DB targets. See `astrapy.constants.Environment` for allowed values.
+        dimension: the number of components for embedding vectors. If not provided, it will be guessed from the embedding driver.
+        metric: the similarity metric to use, one of "dot_product", "euclidean" or "cosine".
+            If omitted, the server default ("cosine") will be used. See also values of `astrapy.constants.VectorMetric`.
+            If the vectors are normalized to unit norm, choosing "dot_product" over cosine yields up to 2x speedup in searches.
+        astra_db_namespace: optional specification of the namespace (in the Astra database) for the data.
+            *Note*: not to be confused with the "namespace" mentioned elsewhere, which is a grouping within this vector store.
+    """
+
+    api_endpoint: str = field(kw_only=True, metadata={"serializable": True})
+    token: Optional[str | TokenProvider] = field(kw_only=True, default=None, metadata={"serializable": False})
+    collection_name: str = field(kw_only=True, metadata={"serializable": True})
+    environment: Optional[str] = field(kw_only=True, default=None, metadata={"serializable": True})
+    dimension: Optional[int] = field(kw_only=True, default=None, metadata={"serializable": True})
+    metric: Optional[str] = field(kw_only=True, default=None, metadata={"serializable": True})
+    astra_db_namespace: Optional[str] = field(default=None, kw_only=True, metadata={"serializable": True})
+
+    collection: Collection = field(init=False)
+
+    def __attrs_post_init__(self) -> None:
+        astrapy = import_optional_dependency("astrapy")
+        if not self.dimension:
+            # auto-compute dimension from the embedding
+            self.dimension = len(self.embedding_driver.embed_string("This is a sample text."))
+        self.collection = (
+            astrapy.DataAPIClient(
+                caller_name="griptape",
+                caller_version=GRIPTAPE_VERSION,
+                environment=self.environment,
+            )
+            .get_database(
+                self.api_endpoint,
+                token=self.token,
+                namespace=self.astra_db_namespace,
+            )
+            .create_collection(
+                name=self.collection_name,
+                dimension=self.dimension,
+                metric=self.metric,
+                indexing=COLLECTION_INDEXING,
+                check_exists=False,
+            )
```

Comment:
Vector Store Drivers shouldn't create any resources, they are intended only for data-plane operations.

Reply:
Umm, I think this point warrants careful examination. Thanks for pointing it out!

If I get it right, the pattern you suggest would unfold invariably as follows:
1. user creates the collection before running the Griptape application (e.g. through the Astra DB Web UI);
2. they run the Griptape application, which finds the collection provisioned already, and starts using it.

If this is something vector store drivers should strictly adhere to, it can be done of course. (my personal opinion is that it would introduce a little more friction in adoption, but I see the cleanliness of this separation of concerns all right).

I guess I am trying to figure out how "negotiable" this requirement is. Admittedly, if the driver has the possibility to create the collection itself, a couple of further problems would be avoided: dimension mismatch (on Astra, users need to specify the vector dimension), optimal indexing setting (tailored to the usage Griptape does of the various document field; such optimization is not available if creating collections through the UI).

Let me know your opinion on this point, and I will adapt to whatever suggestion you have. (in case we end up removing the "create collection" step, I will have to add something to the .md files to instruct the users better around the UI).

Reply:
What you described is correct. Today, we require users to create the collection/index/etc ahead of using the Vector Store Driver. While we could introduce a `create_collection` method, I fear that this would lead to the Drivers feeling a bit inconsistent -- as a user I would assume similar methods exist in other places.

I do think this could be valuable functionality, but I think we should implement for all Drivers at once. I created [an issue](https://github.com/griptape-ai/griptape/issues/1030) to track this.

Reply:
Right, this sounds a very valid and tidy point of view. I will adapt the code, test and examples accordingly, thank you for the prompt reply!

Reply:
Done. The more I look at this, the more it makes sense to me.